src/main/java/io/github/eliux/mega/cmd/ExportInfo.java

```diff
@@ -10,22 +16,44 @@
  */
 public class ExportInfo {
 
+    private static String dateFormat = "EEE, dd MMM yyyy HH:mm:ss Z";
     private static final Pattern LIST_PATTERN =
             Pattern.compile("(?<remotePath>\\S+) \\(.+link: (?<publicLink>http[s]?://mega.nz/\\S*#.+)\\)");
 
+    private static final Pattern EXPORT_PATTERN =
+            Pattern.compile("\\s(?<remotePath>\\S+)(:)\\s(?<publicLink>http[s]?://mega.nz/\\S*#\\S*)( expires at (?<expireDate>.{3}, .{2} .{3} .{4} .{2}:.{2}:.{2} .+))?");
+
     private final String remotePath;
 
     private final String publicLink;
 
+    private Optional<String> expireDate;
+
     public ExportInfo(String remotePath, String publicLink) {
         this.remotePath = remotePath;
         this.publicLink = publicLink;
     }
 
     public static ExportInfo parseExportInfo(String exportInfoStr) {
-        final String[] tokens = exportInfoStr.replace("Exported ", "")
-                .split(": ");
-        return new ExportInfo(tokens[0], tokens[1]);
+        final Matcher matcher = EXPORT_PATTERN.matcher(exportInfoStr);
+
+        if (matcher.find()) {
+            String remotePath = matcher.group("remotePath");
+            String publicLink = matcher.group("publicLink");
+            String expireDate = matcher.group("expireDate");
+
+            if (expireDate == null && exportInfoStr.contains("expires at")) {
+                throw new MegaInvalidExpireDateException(exportInfoStr);
+            }
+            if (expireDate != null) {
```

Comment:
I still need to check if expireDate is not present but exportStr contains "expires at" in case the official format of the date changes.
```
            Optional<String> expireDate = Optional.ofNullable(matcher.group("expireDate"));

            if (!expireDate.isPresent() && exportInfoStr.contains("expires at")) {
                throw new MegaInvalidExpireDateException(exportInfoStr);
            }

            final ExportInfo result = new ExportInfo(remotePath, publicLink);
            expireDate.ifPresent(result::setExpireDate);
```

Reply:
never mind,  i don't actually need this anymore